allennlp/data/dataset_readers/dataset_utils/span_utils.py

```diff
@@ -131,47 +129,48 @@ def bio_tags_to_spans(tag_sequence: List[str],
     return list(spans)
 
 
-class InvalidTagSequence(Exception):
-    pass
-
-
-def bioul_tags_to_spans(tag_sequence: List[str]) -> List[TypedStringSpan]:
+def bioul_tags_to_spans(tag_sequence: List[str],
+                        classes_to_ignore: List[str] = None) -> List[TypedStringSpan]:
     """
-    Given a sequence corresponding to BIOLU tags, extracts spans.
+    Given a sequence corresponding to BIOUL tags, extracts spans.
     Spans are inclusive and can be of zero length, representing a single word span.
     Ill-formed spans are not allowed and will raise ``InvalidTagSequence``.
 
     Parameters
     ----------
     tag_sequence : ``List[str]``, required.
-        The tag sequence encoded in BIOLU, e.g. ["B-PER", "L-PER", "O"].
+        The tag sequence encoded in BIOUL, e.g. ["B-PER", "L-PER", "O"].
+    classes_to_ignore : ``List[str]``, optional (default = None).
+        A list of string class labels `excluding` the bio tag
+        which should be ignored when extracting spans.
 
     Returns
     -------
     spans : ``List[TypedStringSpan]``
         The typed, extracted spans from the sequence, in the format (label, (span_start, span_end)).
     """
     spans = []
-    k = 0
-    while k < len(tag_sequence):
-        label = tag_sequence[k]
+    classes_to_ignore = classes_to_ignore or []
+    index = 0
+    while index < len(tag_sequence):
+        label = tag_sequence[index]
         if label[0] == 'U':
-            spans.append((label.partition('-')[2], (k, k)))
+            spans.append((label.partition('-')[2], (index, index)))
         elif label[0] == 'B':
-            start = k
+            start = index
             while label[0] != 'L':
-                k += 1
-                if k >= len(tag_sequence):
+                index += 1
+                if index >= len(tag_sequence):
                     raise InvalidTagSequence
-                label = tag_sequence[k]
+                label = tag_sequence[index]
                 if not (label[0] == 'I' or label[0] == 'L'):
```

Comment:
I know the code was already like this, but we're inside a `while label[0] != 'L'`, so there's no way `label[0]` could equal `L` here, right?

Reply:
This line checks the next label in the sequence due to the `index += 1` a few lines above inside the loop.

Reply:
👍